examples/step-74/step-laplace-FEIV.cc

```diff
@@ -0,0 +1,1190 @@
+        0.5 * h * (flux_jump_square + penalty * u_jump_square);
+      copy_data_face.values[1] = copy_data_face.values[0];
+    };
+
+    auto copier = [&](const CopyData &copy_data) {
+      if (copy_data.cell_index != numbers::invalid_unsigned_int)
+        estimated_error_per_cell[copy_data.cell_index] += copy_data.value;
+      for (auto &cdf : copy_data.face_data)
+        for (unsigned int j = 0; j < 2; ++j)
+          estimated_error_per_cell[cdf.cell_indices[j]] += cdf.values[j];
+    };
+
+    const unsigned int n_gauss_points = dof_handler.get_fe().degree + 1;
+    QGauss<dim>        quadrature(n_gauss_points);
+    QGauss<dim - 1>    face_quadrature(n_gauss_points);
+
+    UpdateFlags cell_flags =
+      update_hessians | update_quadrature_points | update_JxW_values;
+    UpdateFlags face_flags = update_values | update_gradients |
+                             update_quadrature_points | update_JxW_values |
+                             update_normal_vectors;
+
+    ScratchData scratch_data(
+      mapping, fe, quadrature, cell_flags, face_quadrature, face_flags);
+
+    CopyData cd;
+    MeshWorker::mesh_loop(dof_handler.begin_active(),
+                          dof_handler.end(),
+                          cell_worker,
+                          copier,
+                          scratch_data,
+                          cd,
+                          MeshWorker::assemble_own_cells |
+                            MeshWorker::assemble_own_interior_faces_once |
+                            MeshWorker::assemble_boundary_faces,
+                          boundary_worker,
+                          face_worker);
+  }
+
+
+  template <int dim>
+  double SIPGLaplace<dim>::compute_energy_norm()
+  {
+    typedef decltype(dof_handler.begin_active()) Iterator;
+    const Solution<dim>                          boundary_function(test_case);
+    energy_norm_per_cell.reinit(triangulation.n_active_cells());
+
+    auto cell_worker = [&](const Iterator &cell,
+                           ScratchData &   scratch_data,
+                           CopyData &      copy_data) {
+      const FEValues<dim> &fe_v = scratch_data.reinit(cell);
+
+      copy_data.cell_index = cell->active_cell_index();
+
+      const auto &               q_points   = fe_v.get_quadrature_points();
+      const unsigned int         n_q_points = q_points.size();
+      const std::vector<double> &JxW        = fe_v.get_JxW_values();
+
+      std::vector<Tensor<1, dim>> grad_u(n_q_points);
+      fe_v.get_function_gradients(solution, grad_u);
+
+      std::vector<Tensor<1, dim>> grad_exact(n_q_points);
+      boundary_function.gradient_list(q_points, grad_exact);
+
+      double norm_square = 0;
+      for (unsigned int point = 0; point < n_q_points; ++point)
+        {
+          double tmp = 0.0;
+          for (unsigned int d = 0; d < dim; ++d)
+            {
+              const double diff = grad_u[point][d] - grad_exact[point][d];
+              tmp += diff * diff;
```

Comment:
I think you should be able to write the whole inner loop as ``(grad_u[point] - grad_exact[point]).norm_square()``